llvm/include/llvm/IR/DebugInfoMetadata.h

```diff
@@ -712,31 +712,38 @@ class DIType : public DIScope {
   DIFlags Flags;
   uint64_t SizeInBits;
   uint64_t OffsetInBits;
+  uint32_t NumExtraInhabitants;
```

Comment:
Does the Swift frontend ever use the OffsetInBits? If not, we could store the extra inhabitants in there. (That would be an implementation detail of DIType and DIType would assert that you don't try to set both extra inhabitants and an offset).

Reply:
For everything else (Bitcode, Interface, assembler) leave it as it.

Reply:
Swift uses `OffsetInBits` for `DIDerivedType` but not for `DIBasicType` or `DICompositeType`. `OffsetInBits` is used by other compilers in `DICompositeType` though. I prototyped what you suggested here but I think it'd make things significantly more complex and error prone for little gain. For example, when printing a `DICompositeType` we wouldn't be able to know if the field is an offset or the number of extra inhabitants, unless we add a flag tracking that on `DIType` (or tracking that in `DICompositeType` and making `getOffsetInBits` virtual) which I think would defeat the purpose of sharing the field for both types...